### DISCUSSION TITLE
fp16相关问题

### DISCUSSION THREAD

**Discussion by aso538** [open-mmlab/mmdetection#9269] [General] [answered]
### Prerequisite

- [X] I have searched [the existing and past issues](https://github.com/open-mmlab/mmdetection/issues) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmdetection).

### 💬 Describe the reimplementation questions

config中启用fp16 = dict(loss_scale=dict(init_scale=512))

### Environment

mmdet 2.x
pytorch 1.7

### Additional information

为什么我使用了fp16推理和fp32所用时间是一样的 是因为我的模型是在fp32上训练的吗。以及为什么我fp16训练 显存好像并没有减少，同时更难收敛了。使用的是htc网络，骨干网络替换成了swinv2。显卡是teslaA100

**Comment by mm-assistant** [comment]
We recommend using English or English & Chinese for issues so that we could have broader discussion.

**Comment by ZwwWayne** [comment, marked as answer]
A100 uses tf32 format in computation by default, its speed is already faster than FP32 and compatible with FP16. Therefore, the speed might not be accelerated in some cases. In MMDetection, due to the complexity of the model, many parts are explicitly designed to use FP32 to avoid failure.

**Comment by aso538** [comment]
那么请问teslaT4也是这种情况嘛 我的推理是在T4上做的

**Comment by ZwwWayne** [comment]
T4 does not support inference with TF32.
